obspy/clients/httpproxy.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+"""
+Establish a socket connection through an HTTP proxy.
+Author: Fredrik Østrem <frx.apps@gmail.com>
+License:
+  This code can be used, modified and distributed freely, as long as it is this
+  note containing the original author, the source and this license, is put
+  along with the source code.
+
+J. MacCarthy, modified from https://gist.github.com/frxstrem/4487802
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+from future import standard_library
+from future.utils import native_str
+
+from base64 import b64encode
+import socket
+with standard_library.hooks():
+    import urllib
+    import urlparse
+
+
+def get_proxy_tuple():
+    """
+    Return system http proxy as a urlparse.urlparse tuple or () if unset.
+    """
+    proxydict = urllib.getproxies()
+    proxystr = proxydict.get('http') or proxydict.get('https') or ''
```

Comment:
Based on my reading of `getproxies` docs, I'm not 100% this is correct. The dictionary is scheme->proxy, where the scheme is determined by the URL to which you want to connect. Of course, we're actually connecting to something that's neither, so I'm not sure which is better.

I recall something saying it may be the case that HTTPS proxies are better because they're less likely to mangle any data (because they're transporting encrypted data, they can't change it however they like), but I'm not sure where I read that.